framework/lib/components/toolbox/toolbox.tsx

```diff
@@ -72,13 +83,49 @@ export const Toolbox = ({
           in={ isOpen }
           style={ {
             height: container.h as string,
-            width: toolboxWidth,
+            width: adjustableWidth,
             zIndex: coreZIndex.overlay,
           } }
         >
-          <Flex sx={ { ...container, w: toolboxWidth } } { ...rest } onKeyDown={ handleKeyDown }>
+          <Flex
+            sx={ { ...container, w: adjustableWidth } }
+            { ...rest }
+            onKeyDown={ handleKeyDown }
+            style={ {
+              position: 'relative',
+              overflow: 'hidden',
+            } }
+          >
+            <div
+              style={ {
+                cursor: isResizable ? 'ew-resize' : 'unset',
+                position: 'absolute',
+                top: 0,
+                left: 0,
+                width: '10px',
+                height: '100%',
+                zIndex: 2,
+                userSelect: 'none',
+              } }
+              onMouseDown={ (e) => {
+                if (isResizable) {
```

Comment:
I could not put this into a handler, because of multiple reasons:

1. It makes the `onMouseDown` handler very big (that's usually what you're trying to avoid when writing JSX)
2. It can run in a weird way on slow machines, since you "add" and "remove" handlers every time there's a mouse interaction with the object.

Could you experiment and put it into an outside handler. Something like this:
```
[isDragging, setIsDragging] = boolean

useEffect {
  bind handlers on mount
  unbind handlers on unmount
}
```

To bind handlers, ensure that you use `useRef` hook:
```
const _myRef = useRef()

...

<div ref={ _myRef }>...</div>
```

Reply:
Can you clarify here a bit? Not sure how this optimizes it as I still need the start position

Reply:
@olchyk98 Yes so I talked with Oles an we will leave this be, we can still move the function up like this tho: "handeMouseDown = () => ..." for readibility purposes